src/web/src/utils.js

```diff
@@ -163,6 +163,9 @@ export const withinDuration = (s1, s2) => {
 
 export const generateRequirementsText = (prereqs, coreqs, raw) => {
   let text = [];
+  if (!prereqs && !coreqs) {
+    return "This course has no prerequisites";
```

Comment:
I believe you also need to check that raw is an empty string as well. there are cases where we are unable to parse the prereqs and coreqs